libraries/lib-music-information-retrieval/PowerSpectrumGetter.h

```diff
@@ -29,17 +29,22 @@ class PowerSpectrumGetter
    ~PowerSpectrumGetter();
 
    /*!
-    * @brief Computes the power spectrum of `buffer` into `output`.
-    * @param buffer Input samples of size `fftSize`. Also gets used as
-    * placeholder and gets overwritten, so copy your data elsewhere if you need
-    * it again afterwards.
+    * @brief Returns a pointer to the data to be transformed. User must fill it
+    * with `fftSize` samples before calling `Process()`.
+    */
+   float* GetInputPtr();
+
+   /*!
+    * @brief Computes the power spectrum of data pointed at by `GetInputPtr()`
+    * into `output`.
     * @param output `fftSize / 2 + 1` samples.
     */
-   void operator()(float* buffer, float* output);
+   void Process(float* output);
 
 private:
    const int mFftSize;
    PFFFT_Setup* mSetup;
-   std::vector<float> mWork;
+   float* mData;
+   float* mWork;
```

Comment:
Now that the class has dumb pointers instead of a vector member, it requires the explicit free in the destructor.

So it should also `delete` the copy and assignment operators, to prevent accidental implied copies that would lead to double-free errors.

That's the easy way.  Or for future generality you could define a copy instead, doing the work of separate allocation -- in fact I did need that for this class, in my experimental branch to parallelize it.

And you could define a move constructor and assignment for completeness, or not -- it's not necessary yet.

Reply:
In fact, it was already true because of the `mSetup` member, that copy and assignment should have been `delete`d even before this.  I overlooked it in earlier review.